lms/static/js/spec/student_account/account_settings_fields_spec.js

```diff
@@ -0,0 +1,320 @@
+                    view.message("Do not reset this!");
+                    jasmine.Clock.tick(5000);
+                    expectMessageContains(view, "Do not reset this!");
+                }
+            });
+
+            it("sends a PATCH request when saveAttributes is called", function() {
+
+                requests = AjaxHelpers.requests(this);
+
+                var fieldView = AccountSettingsFieldViews.FieldView;
+                var fieldData = createFieldData(fieldView, {
+                    title: 'Preferred Language',
+                    valueAttribute: 'language',
+                    helpMessage: 'Your preferred language.'
+                })
+
+                var view = new fieldView(fieldData);
+                view.saveAttributes({'language': 'ur'}, {
+                    'headers': {'Priority': 'Urgent'}
+                });
+
+                var request = requests[0];
+                expect(request.method).toBe('PATCH');
+                expect(request.requestHeaders['Content-Type']).toBe('application/merge-patch+json;charset=utf-8');
+                expect(request.requestHeaders['Priority']).toBe('Urgent');
+                expect(request.requestBody).toBe('{"language":"ur"}');
+            });
+
+            it("correctly renders ReadonlyFieldView", function() {
+                var fieldData = createFieldData(AccountSettingsFieldViews.ReadonlyFieldView, {
+                    title: 'Username',
+                    valueAttribute: 'username',
+                    helpMessage: 'This is your login username'
+                });
+                var view = new AccountSettingsFieldViews.ReadonlyFieldView(fieldData).render();
+
+                expectTitleAndMessageToBe(view, fieldData.title, fieldData.helpMessage);
+                expect(view.$('.account-settings-field-value').text().trim()).toBe(MOCK_USERNAME);
+            });
+
+            it("correctly updates ReadonlyFieldView on model update", function() {
+                var fieldData = createFieldData(AccountSettingsFieldViews.ReadonlyFieldView, {
+                    title: 'Username',
+                    valueAttribute: 'username',
+                    helpMessage: 'This is your login username'
+                });
+                var view = new AccountSettingsFieldViews.ReadonlyFieldView(fieldData).render();
+
+                expect(view.$('.account-settings-field-value').text().trim()).toBe(MOCK_USERNAME);
+                view.model.set({'username': 'bookworm'});
+                expect(view.$('.account-settings-field-value').text().trim()).toBe('bookworm');
+            });
+
+            it("correctly renders TextFieldView", function() {
+                var fieldData = createFieldData(AccountSettingsFieldViews.TextFieldView, {
+                    title: 'Full Name',
+                    valueAttribute: 'name',
+                    helpMessage: 'This is used on your edX certificates, and all changes are reviewed.'
```

Comment:
I'm not sure what "changes are reviewed" means here - who does the reviewing? edX? The student? Also, what would that person be checking for - illegal characters, for example?